contents/docker/Docker.md

```diff
@@ -0,0 +1,161 @@
+
+#### 2.5 Docker gives security
+
+From a security point of view, Docker ensures that applications that are running on containers are completely segregated and isolated from each other, granting you complete control over traffic flow and management. No Docker container can look into processes running inside another container. 
+
+This is ideal especially when you dont want Users to exploit vulnerabilities
+between different applications. Docker's isolation ensures that this is not even
+a factor to consider.
+
+### 3.0 When to use docker?
+
+Despite all the abovementioned advantages, it still may not be completely clear
+as to why we need Docker. There are certain situations where it is ideal to make
+use of Docker and 'containerization'.
+
+Some situations you may want to use Docker is as follows:
+
+**Learning new technologies**: To get started with a new tool without spending time on installation and configuration, Docker offers an isolated and disposable environment. Many projects maintain Docker images with their applications already installed and configured. For example, if you want to check out a distributed tracing system like Zipkin, you can have a working setup on localhost simply by running:
+    
+```
+sudo docker run -d -p 9411:9411 openzipkin/zipkin
+```
+
+**Basic use cases**: Pulling images from Docker Hub is also a good solution if your application is basic or standard enough to work with a default Docker image. Cases such as hosting a website using a LAMP stack, using a reverse proxy, or setting up a Minecraft server often have an official or well-supported image available on DockerHub. If the default configuration in these images is acceptable for your needs, then pulling the image can save a lot of time that would otherwise be spent setting up your environment and installing the necessary tools.
+
+**Developer teams**: If you have developers working with different setups, Docker provides a convenient way to have local development environments that closely match the production environment, without needing to ssh into a remote box.
+
+There are many more use cases with Docker. If you face a devOps related problem,
+there is a good chance that docker may provide some solution to it!
+
+### Further Reading / References
+- [Advantages of Docker](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/7.0_release_notes/sect-red_hat_enterprise_linux-7.0_release_notes-linux_containers_with_docker_format-advantages_of_using_docker) 
+- [Introduction to Docker - step-by-step](https://medium.freecodecamp.org/a-beginner-friendly-introduction-to-containers-vms-and-docker-79a9e3e119b) - step by step introduction covering VMs and Containers
+- [Definitions dealing with Docker](https://www.edureka.co/blog/what-is-docker-container) - some important terminology related to Docker
+- [When to use Docker?](https://www.linode.com/docs/applications/containers/when-and-why-to-use-docker/) - some tips on when to go with Docker
```

Comment:
Can add a resource on how docker works, as I think it is important to know how docker is implemented so as to know when it appropriate to use docker etc. 
For example http://docker-saigon.github.io/post/Docker-Internals/